SpatialGDK/Source/SpatialGDK/Public/Schema/Interest.h

```diff
@@ -247,8 +253,8 @@ inline void AddQueryConstraintToQuerySchema(Schema_Object* QueryObject, Schema_F
 
 inline void AddQueryToComponentInterestSchema(Schema_Object* ComponentInterestObject, Schema_FieldId Id, const Query& Query)
 {
-	checkf(!(Query.FullSnapshotResult.IsSet() && Query.ResultComponentIds.Num() > 0),
-		   TEXT("Either full_snapshot_result or a list of result_component_id should be provided. Providing both is invalid."));
+	checkf(!(Query.FullSnapshotResult.IsSet() && Query.ResultComponentSetIds.Num() > 0),
```

Comment:
this would pass if full snapshot result was set but result component IDs had some in the list 
`!(Query.FullSnapshotResult.IsSet() && (Query.ResultComponentSetIds.Num() > 0 || Query.ResultComponentIds.Num() >0))`

Reply:
comment below needs fixing too

Reply:
It looks like the only option that's invalid is setting FullSnapshotResult to false 🤷 
I've updated the check and comment.